### PR TITLE
Fix `deprecated-non-prototype` warnings, pass 1

### PR DESCRIPTION
**Pull request by ribbanya**
Working toward `-requireprotos`.

### CODE REVIEW

File: obj_files.mk [hunk 1]
```diff
@@ -581,7 +581,7 @@ TEXT_O_FILES +=\
     $(BUILD_DIR)/asm/dolphin/mtx/vec.s.o\
 
 TEXT_O_FILES +=\
-    $(BUILD_DIR)/src/dolphin/os/OS.c.o\
+    $(BUILD_DIR)/src/dolphin/os/OSInit.c.o\
```
Comment (r-burns) [inline obj_files.mk:584]: OS.c is the correct file name as provided by debug maps

**Review by r-burns — CHANGES_REQUESTED** [review]

File: src/sysdolphin/baselib/aobj.c [hunk 2]
```diff
@@ -259,45 +259,47 @@ void HSD_AObjFree(HSD_AObj* aobj)
 }
 
 void callbackForeachFunc(struct _HSD_AObj* aobj, void* obj, HSD_Type type,
-                         void (*func)(), AObj_Arg_Type arg_type,
-                         callbackArg* arg)
+                         void* func, AObj_Arg_Type arg_type, callbackArg* arg)
```
Comment (r-burns) [inline src/sysdolphin/baselib/aobj.c:262]: This seems like a case where not specifying the argument types is cleaner. Requiring the explicit casts here adds no useful information

File: obj_files.mk (hunk 1 above)
Comment (ribbanya) [reply obj_files.mk:584]: Well that's fair but we use wine to compile so something has to be done about `os.h` if we want `OS.h`.

File: src/sysdolphin/baselib/aobj.c (hunk 2 above)
Comment (ribbanya) [reply src/sysdolphin/baselib/aobj.c:262]: A small price to pay imo.

Comment (ribbanya) [reply src/sysdolphin/baselib/aobj.c:262]: Just to clarify, `-Wstrict-prototypes` on clang is a warning but `-requireprotos` does not have any flexibility to it. MWCC will abort if it comes across non-compliant code with that option enabled. So I agree the casts suck but I don't think code like that outweighs the benefits.

File: obj_files.mk (hunk 1 above)
Comment (r-burns) [reply obj_files.mk:584]: Hmm that's true. I suppose we could do dolphin/os/OS.h for this TU and dolphin/os.h as the module catchall, but that would require restructuring.

Comment (ribbanya) [reply obj_files.mk:584]: Long-term all of `dolphin` is being outsourced, so personally I don't have a problem with naming this wrong temporarily. But I leave the decision up to you, I was just trying to get it to compile.

File: src/sysdolphin/baselib/aobj.c (hunk 2 above)
Comment (r-burns) [reply src/sysdolphin/baselib/aobj.c:262]: `void (*func)()` is compliant with -requireprotos though?

Comment (ribbanya) [reply src/sysdolphin/baselib/aobj.c:262]: I can double-check but I'm pretty sure it's not.

Comment (r-burns) [reply src/sysdolphin/baselib/aobj.c:262]: Oh I get the issue, `void (*func)()` isn't C99 compliant so you're having trouble using it with clang-tools. That's probably gonna be an issue for a lot of modern tooling. I can deal with the ugliness in that case

**Review by r-burns — APPROVED** [review]

File: src/sysdolphin/baselib/aobj.c (hunk 2 above)
Comment (ribbanya) [reply src/sysdolphin/baselib/aobj.c:262]: Actually no, I elevated the warning to an error in `compile_flags.txt` _because_ `-requireprotos` doesn't allow empty parentheses.

This was like a month ago though so I'm not 100% sure anymore.

Comment (r-burns) [reply src/sysdolphin/baselib/aobj.c:262]: Well I just tested it and does compile with -requireprotos, but a stricter -Werror=strict-prototypes flag is probably a net improvement.

Comment (ribbanya) [reply src/sysdolphin/baselib/aobj.c:262]: I see, good to know, thanks.

**Pull request merged**